scripts/publish.ts

```diff
@@ -10,16 +10,39 @@ import { logging, tags } from '@angular-devkit/core';
 import { spawnSync } from 'child_process';
 import * as semver from 'semver';
 import { packages } from '../lib/packages';
+import { wombat } from '../lib/registries';
 import build from './build';
 
-
 export interface PublishArgs {
   tag?: string;
-  branchCheck?: boolean;
-  versionCheck?: boolean;
+  branchCheck?: string; // Represents a boolean (should be "true" or "false").
```

Comment:
This is a messy situation, I think, due to the dynamic way we parse command line arguments. 
If it is specified in the command line as `--branchCheck` then it'll be a Boolean.
Personally I think keeping it as a boolean makes sense, so we don't have to guess what the possible values are.

Reply:
Ah, I see. The error messages suggested using `--*Check=false`, I thought this was an existing bug where we were getting strings when we expected booleans. Now I see that minimist does support this, it's just that those messages were wrong.

After testing, it seems that `--no-*` is correctly interpreted as a boolean set to `false`. I updated the error messages (and a test) to use `--no-*` and reverted the types to booleans. (The casing is a little weird as it mixes skewer and camel case `--no-branchCheck`, but I think that's a problem for another day).